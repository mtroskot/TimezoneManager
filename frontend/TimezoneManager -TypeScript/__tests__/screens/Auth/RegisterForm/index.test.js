import React from 'react';
import { View } from 'react-native';
import { shallow } from 'enzyme';
import RegisterForm from 'src/screens/Auth/RegisterForm';

describe('RegisterForm wrapper', () => {
  it('renders correctly, form empty, isEdit false', () => {
    const props = {
      submitButtonText: 'Register',
      headerText: 'Register Account',
      registerForm: {
        firstName: '',
        lastName: '',
        emailAddress: '',
        password: '',
        matchingPassword: ''
      },
      errors: {},
      handleInput: jest.fn(),
      handleRegister: jest.fn(),
      isLoading: false,
      isEdit: false
    };

    const wrapper = shallow(<RegisterForm {...props} />);
    expect(wrapper.first().type()).toEqual(View);
    expect(wrapper.find('Text')).toHaveLength(1);
    expect(wrapper.find('Text').props().children).toEqual(props.headerText);
    expect(wrapper.find('Memo(FloatingLabelTextInput)')).toHaveLength(5);
    //first text input
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(0)
        .prop('value')
    ).toEqual(props.registerForm.firstName);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(0)
        .prop('error')
    ).toEqual(undefined);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(0)
        .prop('floatingLabel')
    ).toEqual('First Name');
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(0)
        .prop('returnKeyType')
    ).toEqual('next');
    //2nd text input
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(1)
        .prop('value')
    ).toEqual(props.registerForm.lastName);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(1)
        .prop('error')
    ).toEqual(undefined);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(1)
        .prop('floatingLabel')
    ).toEqual('Last Name');
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(1)
        .prop('returnKeyType')
    ).toEqual('next');
    //3rd text input
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(2)
        .prop('value')
    ).toEqual(props.registerForm.emailAddress);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(2)
        .prop('error')
    ).toEqual(undefined);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(2)
        .prop('floatingLabel')
    ).toEqual('Email Address');
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(2)
        .prop('returnKeyType')
    ).toEqual('next');
    //4th text input
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(3)
        .prop('value')
    ).toEqual(props.registerForm.password);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(3)
        .prop('error')
    ).toEqual(undefined);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(3)
        .prop('floatingLabel')
    ).toEqual('Password');
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(3)
        .prop('returnKeyType')
    ).toEqual('next');
    //5th text input
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(4)
        .prop('value')
    ).toEqual(props.registerForm.matchingPassword);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(4)
        .prop('error')
    ).toEqual(undefined);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(4)
        .prop('floatingLabel')
    ).toEqual('Matching Password');
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(4)
        .prop('returnKeyType')
    ).toEqual('go');
    // button
    expect(wrapper.find('Memo(CustomButton)')).toHaveLength(1);
    expect(
      wrapper
        .find('Memo(CustomButton)')
        .at(0)
        .prop('isLoading')
    ).toEqual(props.isLoading);
    expect(
      wrapper
        .find('Memo(CustomButton)')
        .at(0)
        .prop('text')
    ).toEqual(props.submitButtonText);

    expect(wrapper).toMatchSnapshot();
  });
  it('renders correctly, form empty, isEdit true', () => {
    const props = {
      submitButtonText: 'Register',
      headerText: 'Register Account',
      registerForm: {
        firstName: '',
        lastName: '',
        emailAddress: '',
        password: '',
        matchingPassword: ''
      },
      errors: {},
      handleInput: jest.fn(),
      handleRegister: jest.fn(),
      isLoading: false,
      isEdit: true
    };

    const wrapper = shallow(<RegisterForm {...props} />);
    expect(wrapper.first().type()).toEqual(View);
    expect(wrapper.find('Text')).toHaveLength(1);
    expect(wrapper.find('Text').props().children).toEqual(props.headerText);
    expect(wrapper.find('Memo(FloatingLabelTextInput)')).toHaveLength(3);
    //first text input
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(0)
        .prop('value')
    ).toEqual(props.registerForm.firstName);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(0)
        .prop('error')
    ).toEqual(undefined);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(0)
        .prop('floatingLabel')
    ).toEqual('First Name');
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(0)
        .prop('returnKeyType')
    ).toEqual('next');
    //2nd text input
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(1)
        .prop('value')
    ).toEqual(props.registerForm.lastName);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(1)
        .prop('error')
    ).toEqual(undefined);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(1)
        .prop('floatingLabel')
    ).toEqual('Last Name');
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(1)
        .prop('returnKeyType')
    ).toEqual('next');
    //3rd text input
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(2)
        .prop('value')
    ).toEqual(props.registerForm.emailAddress);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(2)
        .prop('error')
    ).toEqual(undefined);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(2)
        .prop('floatingLabel')
    ).toEqual('Email Address');
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(2)
        .prop('returnKeyType')
    ).toEqual('go');
    // button
    expect(wrapper.find('Memo(CustomButton)')).toHaveLength(1);
    expect(
      wrapper
        .find('Memo(CustomButton)')
        .at(0)
        .prop('isLoading')
    ).toEqual(props.isLoading);
    expect(
      wrapper
        .find('Memo(CustomButton)')
        .at(0)
        .prop('text')
    ).toEqual(props.submitButtonText);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly, form filled, isEdit false', () => {
    const props = {
      submitButtonText: 'Register',
      headerText: 'Register Account',
      registerForm: {
        firstName: '',
        lastName: '',
        emailAddress: '',
        password: '',
        matchingPassword: ''
      },
      errors: {
        firstName: { display: true, errorMessage: 'errorMessage' },
        lastName: { display: true, errorMessage: 'errorMessage' },
        emailAddress: { display: true, errorMessage: 'errorMessage' },
        password: { display: true, errorMessage: 'errorMessage' },
        matchingPassword: { display: true, errorMessage: 'errorMessage' }
      },
      handleInput: jest.fn(),
      handleRegister: jest.fn(),
      isLoading: true,
      isEdit: false
    };

    const wrapper = shallow(<RegisterForm {...props} />);
    expect(wrapper.first().type()).toEqual(View);
    expect(wrapper.find('Text')).toHaveLength(1);
    expect(wrapper.find('Text').props().children).toEqual(props.headerText);
    expect(wrapper.find('Memo(FloatingLabelTextInput)')).toHaveLength(5);
    //first text input
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(0)
        .prop('value')
    ).toEqual(props.registerForm.firstName);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(0)
        .prop('error')
    ).toEqual(props.errors.firstName);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(0)
        .prop('floatingLabel')
    ).toEqual('First Name');
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(0)
        .prop('returnKeyType')
    ).toEqual('next');
    //2nd text input
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(1)
        .prop('value')
    ).toEqual(props.registerForm.lastName);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(1)
        .prop('error')
    ).toEqual(props.errors.lastName);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(1)
        .prop('floatingLabel')
    ).toEqual('Last Name');
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(1)
        .prop('returnKeyType')
    ).toEqual('next');
    //3rd text input
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(2)
        .prop('value')
    ).toEqual(props.registerForm.emailAddress);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(2)
        .prop('error')
    ).toEqual(props.errors.emailAddress);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(2)
        .prop('floatingLabel')
    ).toEqual('Email Address');
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(2)
        .prop('returnKeyType')
    ).toEqual('next');
    //4th text input
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(3)
        .prop('value')
    ).toEqual(props.registerForm.password);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(3)
        .prop('error')
    ).toEqual(props.errors.password);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(3)
        .prop('floatingLabel')
    ).toEqual('Password');
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(3)
        .prop('returnKeyType')
    ).toEqual('next');
    //5th text input
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(4)
        .prop('value')
    ).toEqual(props.registerForm.matchingPassword);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(4)
        .prop('error')
    ).toEqual(props.errors.matchingPassword);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(4)
        .prop('floatingLabel')
    ).toEqual('Matching Password');
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(4)
        .prop('returnKeyType')
    ).toEqual('go');
    // button
    expect(wrapper.find('Memo(CustomButton)')).toHaveLength(1);
    expect(
      wrapper
        .find('Memo(CustomButton)')
        .at(0)
        .prop('isLoading')
    ).toEqual(props.isLoading);
    expect(
      wrapper
        .find('Memo(CustomButton)')
        .at(0)
        .prop('text')
    ).toEqual(props.submitButtonText);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly, form filled, isEdit true', () => {
    const props = {
      submitButtonText: 'Register',
      headerText: 'Register Account',
      registerForm: {
        firstName: '',
        lastName: '',
        emailAddress: '',
        password: '',
        matchingPassword: ''
      },
      errors: {
        firstName: { display: true, errorMessage: 'errorMessage' },
        lastName: { display: true, errorMessage: 'errorMessage' },
        emailAddress: { display: true, errorMessage: 'errorMessage' },
        password: { display: true, errorMessage: 'errorMessage' },
        matchingPassword: { display: true, errorMessage: 'errorMessage' }
      },
      handleInput: jest.fn(),
      handleRegister: jest.fn(),
      isLoading: false,
      isEdit: true
    };

    const wrapper = shallow(<RegisterForm {...props} />);
    expect(wrapper.first().type()).toEqual(View);
    expect(wrapper.find('Text')).toHaveLength(1);
    expect(wrapper.find('Text').props().children).toEqual(props.headerText);
    expect(wrapper.find('Memo(FloatingLabelTextInput)')).toHaveLength(3);
    //first text input
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(0)
        .prop('value')
    ).toEqual(props.registerForm.firstName);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(0)
        .prop('error')
    ).toEqual(props.errors.firstName);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(0)
        .prop('floatingLabel')
    ).toEqual('First Name');
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(0)
        .prop('returnKeyType')
    ).toEqual('next');
    //2nd text input
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(1)
        .prop('value')
    ).toEqual(props.registerForm.lastName);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(1)
        .prop('error')
    ).toEqual(props.errors.lastName);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(1)
        .prop('floatingLabel')
    ).toEqual('Last Name');
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(1)
        .prop('returnKeyType')
    ).toEqual('next');
    //3rd text input
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(2)
        .prop('value')
    ).toEqual(props.registerForm.emailAddress);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(2)
        .prop('error')
    ).toEqual(props.errors.emailAddress);
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(2)
        .prop('floatingLabel')
    ).toEqual('Email Address');
    expect(
      wrapper
        .find('Memo(FloatingLabelTextInput)')
        .at(2)
        .prop('returnKeyType')
    ).toEqual('go');
    // button
    expect(wrapper.find('Memo(CustomButton)')).toHaveLength(1);
    expect(
      wrapper
        .find('Memo(CustomButton)')
        .at(0)
        .prop('isLoading')
    ).toEqual(props.isLoading);
    expect(
      wrapper
        .find('Memo(CustomButton)')
        .at(0)
        .prop('text')
    ).toEqual(props.submitButtonText);

    expect(wrapper).toMatchSnapshot();
  });
});

describe('interaction', () => {
  const props = {
    submitButtonText: 'Register',
    headerText: 'Register Account',
    registerForm: {
      firstName: '',
      lastName: '',
      emailAddress: '',
      password: '',
      matchingPassword: ''
    },
    errors: {
      firstName: { display: true, errorMessage: 'errorMessage' },
      lastName: { display: true, errorMessage: 'errorMessage' },
      emailAddress: { display: true, errorMessage: 'errorMessage' },
      password: { display: true, errorMessage: 'errorMessage' },
      matchingPassword: { display: true, errorMessage: 'errorMessage' }
    },
    handleInput: jest.fn(),
    handleRegister: jest.fn(),
    isLoading: false,
    isEdit: false
  };

  const wrapper = shallow(<RegisterForm {...props} />);

  it('calling handleRegister,handleInput, should trigger callbacks', () => {
    wrapper
      .find('Memo(FloatingLabelTextInput)')
      .at(0)
      .prop('onChangeText')();
    wrapper
      .find('Memo(FloatingLabelTextInput)')
      .at(1)
      .prop('onChangeText')();
    wrapper
      .find('Memo(FloatingLabelTextInput)')
      .at(2)
      .prop('onChangeText')();
    wrapper
      .find('Memo(FloatingLabelTextInput)')
      .at(3)
      .prop('onChangeText')();
    wrapper
      .find('Memo(FloatingLabelTextInput)')
      .at(4)
      .prop('onChangeText')();
    wrapper
      .find('Memo(FloatingLabelTextInput)')
      .at(4)
      .prop('onSubmitEditing')();
    wrapper
      .find('Memo(CustomButton)')
      .at(0)
      .prop('onPress')();
    expect(props.handleInput).toHaveBeenCalledTimes(5);
    expect(props.handleRegister).toHaveBeenCalledTimes(2);
  });
});
