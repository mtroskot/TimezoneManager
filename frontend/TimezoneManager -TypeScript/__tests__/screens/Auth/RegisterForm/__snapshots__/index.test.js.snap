// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegisterForm wrapper renders correctly, form empty, isEdit false 1`] = `
<View>
  <View
    style={
      Object {
        "marginVertical": 88.8157894736842,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#334144",
          "fontFamily": "Ubuntu-Medium",
          "fontSize": 43.421052631578945,
          "marginBottom": 19.736842105263158,
          "textAlign": "center",
        }
      }
    >
      Register Account
    </Text>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flexDirection": "row",
        "width": "90%",
      }
    }
  >
    <Memo(FloatingLabelTextInput)
      autoCapitalize="words"
      customContainerStyle={
        Object {
          "flex": 1,
        }
      }
      floatingLabel="First Name"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholderTextColor="#949EA0"
      returnKeyType="next"
      testID="FIRST_NAME_INPUT"
      value=""
    />
    <Memo(FloatingLabelTextInput)
      autoCapitalize="words"
      customContainerStyle={
        Object {
          "flex": 1,
          "marginLeft": 19.736842105263158,
        }
      }
      floatingLabel="Last Name"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholderTextColor="#949EA0"
      returnKeyType="next"
      testID="LAST_NAME_INPUT"
      textInputRef={
        Object {
          "current": null,
        }
      }
      value=""
    />
  </View>
  <Memo(FloatingLabelTextInput)
    floatingLabel="Email Address"
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholderTextColor="#949EA0"
    returnKeyType="next"
    testID="EMAIL_INPUT"
    textInputRef={
      Object {
        "current": null,
      }
    }
    value=""
  />
  <Memo(FloatingLabelTextInput)
    floatingLabel="Password"
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholderTextColor="#949EA0"
    returnKeyType="next"
    secureTextEntry={true}
    testID="PASSWORD_INPUT"
    textInputRef={
      Object {
        "current": null,
      }
    }
    value=""
  />
  <Memo(FloatingLabelTextInput)
    floatingLabel="Matching Password"
    onChangeText={[Function]}
    onSubmitEditing={[MockFunction]}
    placeholderTextColor="#949EA0"
    returnKeyType="go"
    secureTextEntry={true}
    testID="MATCHING_PASSWORD_INPUT"
    textInputRef={
      Object {
        "current": null,
      }
    }
    value=""
  />
  <Memo(CustomButton)
    isLoading={false}
    loaderStyle={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "#EAA79E",
          "borderRadius": 9.868421052631579,
          "marginTop": 19.736842105263158,
          "paddingHorizontal": 39.473684210526315,
          "paddingVertical": 15,
          "width": "90%",
        },
        Object {
          "backgroundColor": "transparent",
        },
      ]
    }
    onPress={[MockFunction]}
    tOpacityStyle={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#EAA79E",
        "borderRadius": 9.868421052631579,
        "marginTop": 19.736842105263158,
        "paddingHorizontal": 39.473684210526315,
        "paddingVertical": 15,
        "width": "90%",
      }
    }
    text="Register"
    textStyle={
      Object {
        "color": "#fff",
        "fontFamily": "Ubuntu-Medium",
        "fontSize": 31.57894736842105,
        "textAlign": "center",
      }
    }
  />
</View>
`;

exports[`RegisterForm wrapper renders correctly, form empty, isEdit true 1`] = `
<View>
  <View
    style={
      Object {
        "marginVertical": 88.8157894736842,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#334144",
          "fontFamily": "Ubuntu-Medium",
          "fontSize": 43.421052631578945,
          "marginBottom": 19.736842105263158,
          "textAlign": "center",
        }
      }
    >
      Register Account
    </Text>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flexDirection": "row",
        "width": "90%",
      }
    }
  >
    <Memo(FloatingLabelTextInput)
      autoCapitalize="words"
      customContainerStyle={
        Object {
          "flex": 1,
        }
      }
      floatingLabel="First Name"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholderTextColor="#949EA0"
      returnKeyType="next"
      testID="FIRST_NAME_INPUT"
      value=""
    />
    <Memo(FloatingLabelTextInput)
      autoCapitalize="words"
      customContainerStyle={
        Object {
          "flex": 1,
          "marginLeft": 19.736842105263158,
        }
      }
      floatingLabel="Last Name"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholderTextColor="#949EA0"
      returnKeyType="next"
      testID="LAST_NAME_INPUT"
      textInputRef={
        Object {
          "current": null,
        }
      }
      value=""
    />
  </View>
  <Memo(FloatingLabelTextInput)
    floatingLabel="Email Address"
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholderTextColor="#949EA0"
    returnKeyType="go"
    testID="EMAIL_INPUT"
    textInputRef={
      Object {
        "current": null,
      }
    }
    value=""
  />
  <Memo(CustomButton)
    isLoading={false}
    loaderStyle={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "#EAA79E",
          "borderRadius": 9.868421052631579,
          "marginTop": 19.736842105263158,
          "paddingHorizontal": 39.473684210526315,
          "paddingVertical": 15,
          "width": "90%",
        },
        Object {
          "backgroundColor": "transparent",
        },
      ]
    }
    onPress={[MockFunction]}
    tOpacityStyle={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#EAA79E",
        "borderRadius": 9.868421052631579,
        "marginTop": 19.736842105263158,
        "paddingHorizontal": 39.473684210526315,
        "paddingVertical": 15,
        "width": "90%",
      }
    }
    text="Register"
    textStyle={
      Object {
        "color": "#fff",
        "fontFamily": "Ubuntu-Medium",
        "fontSize": 31.57894736842105,
        "textAlign": "center",
      }
    }
  />
</View>
`;

exports[`RegisterForm wrapper renders correctly, form filled, isEdit false 1`] = `
<View>
  <View
    style={
      Object {
        "marginVertical": 88.8157894736842,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#334144",
          "fontFamily": "Ubuntu-Medium",
          "fontSize": 43.421052631578945,
          "marginBottom": 19.736842105263158,
          "textAlign": "center",
        }
      }
    >
      Register Account
    </Text>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flexDirection": "row",
        "width": "90%",
      }
    }
  >
    <Memo(FloatingLabelTextInput)
      autoCapitalize="words"
      customContainerStyle={
        Object {
          "flex": 1,
        }
      }
      error={
        Object {
          "display": true,
          "errorMessage": "errorMessage",
        }
      }
      floatingLabel="First Name"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholderTextColor="#949EA0"
      returnKeyType="next"
      testID="FIRST_NAME_INPUT"
      value=""
    />
    <Memo(FloatingLabelTextInput)
      autoCapitalize="words"
      customContainerStyle={
        Object {
          "flex": 1,
          "marginLeft": 19.736842105263158,
        }
      }
      error={
        Object {
          "display": true,
          "errorMessage": "errorMessage",
        }
      }
      floatingLabel="Last Name"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholderTextColor="#949EA0"
      returnKeyType="next"
      testID="LAST_NAME_INPUT"
      textInputRef={
        Object {
          "current": null,
        }
      }
      value=""
    />
  </View>
  <Memo(FloatingLabelTextInput)
    error={
      Object {
        "display": true,
        "errorMessage": "errorMessage",
      }
    }
    floatingLabel="Email Address"
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholderTextColor="#949EA0"
    returnKeyType="next"
    testID="EMAIL_INPUT"
    textInputRef={
      Object {
        "current": null,
      }
    }
    value=""
  />
  <Memo(FloatingLabelTextInput)
    error={
      Object {
        "display": true,
        "errorMessage": "errorMessage",
      }
    }
    floatingLabel="Password"
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholderTextColor="#949EA0"
    returnKeyType="next"
    secureTextEntry={true}
    testID="PASSWORD_INPUT"
    textInputRef={
      Object {
        "current": null,
      }
    }
    value=""
  />
  <Memo(FloatingLabelTextInput)
    error={
      Object {
        "display": true,
        "errorMessage": "errorMessage",
      }
    }
    floatingLabel="Matching Password"
    onChangeText={[Function]}
    onSubmitEditing={[MockFunction]}
    placeholderTextColor="#949EA0"
    returnKeyType="go"
    secureTextEntry={true}
    testID="MATCHING_PASSWORD_INPUT"
    textInputRef={
      Object {
        "current": null,
      }
    }
    value=""
  />
  <Memo(CustomButton)
    isLoading={true}
    loaderStyle={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "#EAA79E",
          "borderRadius": 9.868421052631579,
          "marginTop": 19.736842105263158,
          "paddingHorizontal": 39.473684210526315,
          "paddingVertical": 15,
          "width": "90%",
        },
        Object {
          "backgroundColor": "transparent",
        },
      ]
    }
    onPress={[MockFunction]}
    tOpacityStyle={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#EAA79E",
        "borderRadius": 9.868421052631579,
        "marginTop": 19.736842105263158,
        "paddingHorizontal": 39.473684210526315,
        "paddingVertical": 15,
        "width": "90%",
      }
    }
    text="Register"
    textStyle={
      Object {
        "color": "#fff",
        "fontFamily": "Ubuntu-Medium",
        "fontSize": 31.57894736842105,
        "textAlign": "center",
      }
    }
  />
</View>
`;

exports[`RegisterForm wrapper renders correctly, form filled, isEdit true 1`] = `
<View>
  <View
    style={
      Object {
        "marginVertical": 88.8157894736842,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#334144",
          "fontFamily": "Ubuntu-Medium",
          "fontSize": 43.421052631578945,
          "marginBottom": 19.736842105263158,
          "textAlign": "center",
        }
      }
    >
      Register Account
    </Text>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "flexDirection": "row",
        "width": "90%",
      }
    }
  >
    <Memo(FloatingLabelTextInput)
      autoCapitalize="words"
      customContainerStyle={
        Object {
          "flex": 1,
        }
      }
      error={
        Object {
          "display": true,
          "errorMessage": "errorMessage",
        }
      }
      floatingLabel="First Name"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholderTextColor="#949EA0"
      returnKeyType="next"
      testID="FIRST_NAME_INPUT"
      value=""
    />
    <Memo(FloatingLabelTextInput)
      autoCapitalize="words"
      customContainerStyle={
        Object {
          "flex": 1,
          "marginLeft": 19.736842105263158,
        }
      }
      error={
        Object {
          "display": true,
          "errorMessage": "errorMessage",
        }
      }
      floatingLabel="Last Name"
      onChangeText={[Function]}
      onSubmitEditing={[Function]}
      placeholderTextColor="#949EA0"
      returnKeyType="next"
      testID="LAST_NAME_INPUT"
      textInputRef={
        Object {
          "current": null,
        }
      }
      value=""
    />
  </View>
  <Memo(FloatingLabelTextInput)
    error={
      Object {
        "display": true,
        "errorMessage": "errorMessage",
      }
    }
    floatingLabel="Email Address"
    onChangeText={[Function]}
    onSubmitEditing={[Function]}
    placeholderTextColor="#949EA0"
    returnKeyType="go"
    testID="EMAIL_INPUT"
    textInputRef={
      Object {
        "current": null,
      }
    }
    value=""
  />
  <Memo(CustomButton)
    isLoading={false}
    loaderStyle={
      Array [
        Object {
          "alignSelf": "center",
          "backgroundColor": "#EAA79E",
          "borderRadius": 9.868421052631579,
          "marginTop": 19.736842105263158,
          "paddingHorizontal": 39.473684210526315,
          "paddingVertical": 15,
          "width": "90%",
        },
        Object {
          "backgroundColor": "transparent",
        },
      ]
    }
    onPress={[MockFunction]}
    tOpacityStyle={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#EAA79E",
        "borderRadius": 9.868421052631579,
        "marginTop": 19.736842105263158,
        "paddingHorizontal": 39.473684210526315,
        "paddingVertical": 15,
        "width": "90%",
      }
    }
    text="Register"
    textStyle={
      Object {
        "color": "#fff",
        "fontFamily": "Ubuntu-Medium",
        "fontSize": 31.57894736842105,
        "textAlign": "center",
      }
    }
  />
</View>
`;
